neuropixels_data_sep_2020/extractors/h5extractors/h5sortingextractorv1.py

```diff
@@ -0,0 +1,52 @@
+from spikeextractors import SortingExtractor
+import numpy as np
+
+# import h5py
+import warnings
+warnings.simplefilter(action='ignore', category=FutureWarning)
+import h5py
+warnings.resetwarnings()
+
+class H5SortingExtractorV1(SortingExtractor):
+    def __init__(self, h5_path):
+        SortingExtractor.__init__(self)
+        self._h5_path = h5_path
+        self._loaded_spike_trains = {}
+
+        with h5py.File(self._h5_path, 'r') as f:
+            self._unit_ids = np.array(f.get('unit_ids'))
+            self._sampling_frequency = np.array(f.get('sampling_frequency'))[0]
+
+    def get_unit_ids(self):
+        return self._unit_ids.tolist()
+
+    def get_unit_spike_train(self, unit_id, start_frame=None, end_frame=None):
+        if unit_id not in self._loaded_spike_trains:
+            with h5py.File(self._h5_path, 'r') as f:
+                self._loaded_spike_trains[unit_id] = np.array(f.get(f'unit_spike_trains/{unit_id}'))
+
+        x = self._loaded_spike_trains[unit_id]
+
+        if start_frame is None:
+            start_frame = 0
+        
+        if (start_frame == 0) and (end_frame is None):
+            return x
+        
+        assert end_frame is not None
```

Comment:
Any reason not to just default `end_frame = x[-1]`? Like assume they want to go to the end?

Reply:
In a sorting extractor, the duration of the recording is not known. So that's why i did it that way. But you could just assume it is Inf, and that would work. So I'm not opposed to changing it at some point.